x/cron/keeper/keeper.go

```diff
@@ -67,13 +69,14 @@ func (k *Keeper) Logger(ctx sdk.Context) log.Logger {
 // ExecuteReadySchedules gets all schedules that are due for execution (with limit that is equal to Params.Limit)
 // and executes messages in each one
 func (k *Keeper) ExecuteReadySchedules(ctx sdk.Context, executionStage types.ExecutionStage) {
-	telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), LabelExecuteReadySchedules)
+	startTime := time.Now()
 	schedules := k.getSchedulesReadyForExecution(ctx, executionStage)
 
 	for _, schedule := range schedules {
 		err := k.executeSchedule(ctx, schedule)
 		recordExecutedSchedule(err, schedule)
 	}
+	telemetry.ModuleMeasureSince(types.ModuleName, startTime, LabelExecuteReadySchedules)
```

Comment:
I think just adding `defer` resolves the problem and it's more cleaner from my perspective (Cosmos SDK [uses](https://github.com/neutron-org/cosmos-sdk/blob/c3892cfa7e2e3dc2295209862b58f1e203a18d07/x/slashing/abci.go#L17) `defer` as well):
https://go.dev/play/p/Eof5hpEyxQX